controller/volume_controller.go

```diff
@@ -2853,7 +2853,16 @@ func (vc *VolumeController) checkForAutoDetachment(v *longhorn.Volume, e *longho
 	}
 
 	if v.Status.ExpansionRequired {
-		return nil
+		if v.Status.CurrentNodeID == "" {
+			return nil
+		}
+
+		_, err := vc.ds.GetNodeRO(v.Status.CurrentNodeID)
+		if err == nil || (err != nil && !datastore.ErrorIsNotFound(err)) {
+			return nil
+		}
+
+		v.Status.CurrentNodeID = ""
```

Comment:
Do we need to return after this setting?

Reply:
Updated. Thank you,